### PR TITLE
Create Todo Item

### PR DESCRIPTION
**Pull request by rahulbasrani**
## Description

Create API for todo task POST /todos

## Database schema changes

Model Todo add.
Schema:
{
title: { type: String, required: true }
},
{
collection: 'todos',
timestamps: {
createdAt: 'createdAt',
updatedAt: 'updatedAt',
},
}

### Automated test cases added

- POST /todos
  input: {title: 'Testing title new todo'}
  output: success

- POST /todos
  input: {title: ''}
  output: fail

### Manual test cases run

NA

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Description
```
Comment (vindru) [inline .github/pull_request_template.md:1]: PR template should not be edited, please replace this with default template and add description on PR created directly.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (vindru) [inline .prettierrc]: add eof

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,7 +0,0 @@
-{
```
Comment (vindru) [inline .vscode/settings.json:1]: gitignore this

File: src/storage/mongoose/todo.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { model, Schema } from "mongoose";
+
+const todoSchema: Schema = new Schema(
+  {
+    title: { type: String, required: true },
+    isActive: { type: Boolean, default: true },
```
Comment (vindru) [inline src/storage/mongoose/todo.ts]: isActive is not being used right now, is there any reason its added here?

File: test/mocks/in-memory-mongo-store.ts [hunk 5]
```diff
@@ -8,8 +9,7 @@ export class InMemoryMongoStore extends Mongoose.MongoStore {
   }
 
   public async connect(): Promise<M> {
-    const mongod = new MongoMemoryServer();
-    const uri = await mongod.getConnectionString();
+    const uri: string = config.get('mongoDb.uri');
```
Comment (vindru) [inline test/mocks/in-memory-mongo-store.ts]: use this here :  const uri = await mongod.getUri();

**Review by vindru — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (rahulbasrani) [reply .github/pull_request_template.md:1]: sure

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,4 @@
-*.DS_Store
 node_modules/
 dist/
 config/local.json
+.vscode
```
Comment (vindru) [inline .gitignore:4]: add eof

**Review by vindru — APPROVED** [review]

**Pull request merged**